var/spack/repos/builtin/packages/ucc/package.py

```diff
@@ -55,5 +54,26 @@ def configure_args(self):
         args = []
         args.extend(self.with_or_without("cuda", activation_value="prefix"))
         args.extend(self.with_or_without("nccl", activation_value="prefix"))
-        # args.extend(self.with_or_without("rccl", activation_value="prefix"))
+        if self.spec.satisfies("+rocm"):
+            cppflags = " ".join(
+                "-I" + include_dir
+                for include_dir in (
+                    self.spec["hip"].prefix.include,
+                    self.spec["hip"].prefix.include.hip,
+                    self.spec["hsa-rocr-dev"].prefix.include.hsa,
+                )
+            )
+            ldflags = " ".join(
+                "-L" + library_dir
+                for library_dir in (
+                    self.spec["hip"].prefix.lib,
+                    self.spec["hsa-rocr-dev"].prefix.lib,
+                )
+            )
+            args.extend(["CPPFLAGS=" + cppflags, "LDFLAGS=" + ldflags])
+            args.append("--with-rocm=" + self.spec["hip"].prefix)
+            args.append("--with-ucx=" + self.spec["ucx"].prefix)
+        else:
+            args.append("--without-rocm")
+        args.extend(self.with_or_without("rccl", activation_value="prefix"))
```

Comment:
Is it correct to have this outside the conditional above?

Reply:
It shouldn't matter since rccl can't be set without rocm being set, but I've placed it inside the condition anyways.